test/functional/mempool_reorg.py

```diff
@@ -106,8 +106,45 @@ def run_test(self):
 
         self.log.info("Check that the mempool is empty")
         assert_equal(set(self.nodes[0].getrawmempool()), set())
+        assert_equal(set(self.nodes[1].getrawmempool()), set())
         self.sync_all()
 
 
+        self.log.info("Simulate natural reorg")
+        self.connect_nodes(0, 1)
+        assert_equal(self.nodes[0].getblockcount(), self.nodes[1].getblockcount())
+        assert_equal(set(self.nodes[0].getrawmempool()), set(self.nodes[0].getrawmempool()))
```

Comment:
nit: Why are the asserts needed? The implicit `self.sync_all` in the next line should assert this already.

Reply:
removed